WcaOnRails/app/assets/javascripts/competitions.js

```diff
@@ -99,3 +99,32 @@ onPage('competitions#index', function() {
     }, 0);
   });
 });
+
+onPage("competitions#show_all_results", function() {
+  /* Show tbody for the given event. */
+  function showResultsFor(eventId) {
+    var $results = $('.one-event');
+    $results.hide();
+    $results.filter('.event-' + eventId).show();
+  }
+
+  /* Handle events selector for results by event. */
+  $('.event-selector input[type="radio"]').on('change', function() {
+    var eventId = $(this).val();
+    showResultsFor(eventId);
+    $.setUrlParams({ event:  eventId });
+  });
+
+  if(location.hash) {
+    /* Support old URLs with hash indicating an event id. */
+    $('.event-selector #radio-' + location.hash.slice(2)).click();
+  } else {
+    var $checked = $('.event-selector input[checked="checked"]');
+    if ($checked.length != 0) {
+      showResultsFor($checked.val());
+    } else {
+      var $toCheck = $('.event-selector input:first');
```

Comment:
This got me thinking: what happens if the user has javascript disabled? Does something sane happen? It's ok if not, mostly just curious.

Reply:
This wouldn't work at all with javascript disabled -- no results would be shown.  All of the results are hidden by CSS, and shown in javascript (but that doesn't happen without javascript).

That said, it's pretty hard to even get to this page with javascript disabled, since the menus on top don't work at all.  If we think it's worth supporting browsers with javascript disabled, I think it'd be a decent amount of work to make the site functional.